models/request_result.rb

```diff
@@ -46,12 +46,20 @@ def self.already_sent_error?(message_hash)
     (hash.is_a? Hash) && (hash["description"].is_a? String) && error_list.any? {|error| hash["description"].include?(error)}
   end
 
+  def self.get_patron_holds(patron)
+    begin
+      sierra_request = SierraRequest.build_new_sierra_request({})
+      sierra_request.assign_bearer
+      sierra_request.get_holds(patron)
+    rescue Error => e
+      CustomLogger.new("level" => "ERROR", "message" => "Unable to get holds for patron: #{e.message}")
+    end
+  end
+
   def self.patron_already_has_hold?(hold_request)
     patron = hold_request["data"]["patron"]
     record = hold_request["data"]["record"]
-    sierra_request = SierraRequest.new({})
```

Comment:
wondering if you need to do the whole "build_new_sierra_request" if you're just testing if the patron already has the hold.

Reply:
As far as I can tell the bug this branch is responding to was caused by `sierra_request` not having a base url and trying to hit a nonsense endpoint. Just using `.new` doesn't initialize the base url but `build_new_sierra_request` does.

Reply:
Ah. I see. In that case, I'd suggest adding this line right after line 52, then. 

`sierra_request.base_request_url = ENV['SIERRA_URL']`

Reason being I think the .build_new_sierra_request does some logging that I don't think you want right here.

Reply:
Changed